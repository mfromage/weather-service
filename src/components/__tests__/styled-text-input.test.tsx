import { fireEvent, render } from '@testing-library/react-native';
import StyledTextInput from '@/components/styled-text-input';

describe('StyledTextInput', () => {
  const setup = (props = {}) => {
    const testId = 'styled-text-input-test-id';
    const defaultProps = {
      placeholder: 'type something here...',
      onChangeText: jest.fn(),
    };
    const utils = render(
      <StyledTextInput {...defaultProps} {...props} testID={testId} />,
    );
    const styledTextInput = utils.getByTestId(testId);
    return { ...utils, styledTextInput };
  };
  it('render with given props', () => {
    const placeholder = 'type nothing here...';
    const { getByPlaceholderText } = setup({ placeholder });
    expect(getByPlaceholderText(placeholder)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  it('call onChangeText', () => {
    const onChangeText = jest.fn();
    const { styledTextInput } = setup({ onChangeText });

    fireEvent.changeText(styledTextInput, 'hello');
    expect(onChangeText).toHaveBeenCalledWith('hello');
  });
});
