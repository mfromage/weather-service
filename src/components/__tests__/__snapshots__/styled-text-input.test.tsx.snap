// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextInput matches the snapshot 1`] = `
<TextInput
  autoCorrect={false}
  onChangeText={[MockFunction]}
  placeholder="type something here..."
  style={
    [
      {
        "borderColor": "lightgray",
        "borderRadius": 12,
        "borderWidth": 0.5,
        "height": 40,
        "padding": 12,
      },
      undefined,
    ]
  }
  testID="styled-text-input-test-id"
/>
`;
